capella_ros_tools/exporter.py

```diff
@@ -2,83 +2,625 @@
+                    cls_data.literals.append(
+                        LiteralData(
+                            type_name,
+                            f"{prop_name.upper()}_{self.make_snake_case(val.name).upper()}",
+                            val.value.value,
+                            self._make_doc_str(val.description),
+                        )
+                    )
+            else:
+                type_name = self._make_type_name(_type)
+        elif isinstance(_type, information.Class):
+            pkg = ""
+            build_in = False
+            if _type.uuid not in pkg_cls_uuids:
+                if cls_pkg := class_package_mapping.get(_type.uuid):
+                    if cls_pkg != current_pkg:
+                        pkg_dependencies.add(cls_pkg)
+                        build_in = cls_pkg in self.build_ins
+                        if not build_in:
+                            cls_pkg += self.pkg_postfix
+                        pkg = f"{cls_pkg}/"
+                else:
+                    logger.error(
+                        "Class %s was referenced in %s, but not found",
+                        _type.name,
+                        cls.name,
+                    )
+
+            type_name = pkg + (
+                _type.name if build_in else self.make_camel_case(_type.name)
+            )
+        else:
+            logger.error("Unknown type for property %s of class %s")
+            type_name = "unknown"
+        return type_name
+
+    def _collect_build_in_classes(self) -> dict[str, str]:
+        cls_to_pkg_mapping: dict[str, str] = {}
+        for pkg_name, uuid in self.build_ins.items():
+            for cls in self._get_package_classes(self.model.by_uuid(uuid)):
+                cls_to_pkg_mapping[cls.uuid] = pkg_name
+
+        return cls_to_pkg_mapping
+
+    def prepare_export_data(
+        self,
+    ) -> tuple[dict[str, list[ClassData]], dict[str, set[str]]]:
+        """Collect export data for all defined packages."""
+        package_class_mapping = self._collect_pure_packages()
+        package_class_mapping |= self.custom_pkg
+        class_package_mapping = self._collect_build_in_classes()
+        class_package_mapping |= {
+            cls.uuid: pkg_name
+            for pkg_name, classes in package_class_mapping.items()
+            for cls in classes
+        }
+        dependency_classes = self._get_missing_dependencies(
+            package_class_mapping, class_package_mapping
+        )
+        seen_classes = set()
+        duplicate_classes_uuids = set()
+        multi_dependency_classes = []
+        for clss in dependency_classes.values():
+            for cls in clss:
+                if cls.uuid in seen_classes:
+                    if cls.uuid not in duplicate_classes_uuids:
+                        multi_dependency_classes.append(cls)
+                    duplicate_classes_uuids.add(cls.uuid)
+                else:
+                    seen_classes.add(cls.uuid)
```

Comment:
This is a little bit too nested for my taste. We can at least easily get rid of one level of indentation by inverting the first conditional:

```suggestion
        for clss in dependency_classes.values():
            for cls in clss:
                if cls.uuid not in seen_classes:
                    seen_classes.add(cls.uuid)
                elif cls.uuid not in duplicate_classes_uuids:
                    duplicate_classes_uuids.add(cls.uuid)
                    multi_dependency_classes.append(cls)
```